### ISSUE TITLE
accumulo-cluster start here

### ISSUE THREAD

**Issue by ivakegg** [apache/accumulo#5196]
If one types ..../accumulo-cluster start here, it should fail because "here" is not a valid argument.  This is a common typo that results in the entire cluster being started as opposed to the desired action of start-here.  Please add some addition argument checking on that command.

accumulo version 2.1.3 +

**Comment by ddanielr** [comment]
I believe the new getopt parsing in #5174 will fix this. 
Can test and verify once that PR is merged in.

**Comment by ddanielr** [comment]
Tested on the current branch for #5174. 
It does not catch the issue as we currently support passing extra args at the end of accumulo-cluster. 

We have at least three different ways to handle the situation.

The first is to support a mis-typing of start here or stop here and fix it around line 586. 
Probably should also add a check to do if $@ is equal to 1. 
```
# Handle missing hyphens
if [[ "$1" == "here" ]]; then
   cmd="$cmd-$1"
   shift
fi
```

The second would be to dis-allow it entirely in by checking if "$@" is greater than zero at the end of parse_args and not supporting additional values at all when calling `accumulo-cluster`. 

That would solve the error cases of missing the hyphen and also misspelling the word. `stop heer` or `stop hre`. 

The third would be to not support a default case of `stop` or `start` and always require at least one argument flag to be set i.e `start --all`  `stop --all`. or `start --local` `stop --local`

Option 1 seems extremely limited in scope and wouldn't prevent other error conditions. 
Option 2 would prevent error conditions but would also remove a current ability we support.
Option 3 pushes users to be more direct in their commands, but would cause a breakage for any current SOPs for starting/stopping accumulo. 

@ctubbsii What do you think of the current options and can you think of any other code-based alternatives?

**Comment by ctubbsii** [comment]
@ddanielr wrote:
> What do you think of the current options ... ?

1. :-1: this requires us to mind-read users to protect them from themselves and seems like a losing game for everybody
2. :+1: this seems best because it directly addresses the issue by more thoroughly vetting the user input; I'm not sure what current ability you're referring to; it looks like we're just throwing the rest of the args away at the end of `parse_args`, so I don't think we'd lose anything with this
3. :-1: this doesn't directly address the problem of users' input not being fully validated, and only narrowly fixes the situation indirectly via a breaking change that imposes a new option on users

> can you think of any other code-based alternatives?

I did have an idea to combine the two sections that parse parameters, so instead of pulling the command off before `getopt`, we do it after. That'd be the best place to do any validation of the non-option parameters, like your **Option 2**, and make sure there is exactly one (for the command to be executed). A beneficial side effect of this is that the command would no longer need to be first on the command-line. You could do things like: `accumulo-cluster --all start`. This suggestion doesn't replace your **Option 2**, but it complements it.

**Comment by dlmarion** [comment]
I too am confused on what capability we lose if we implement 2. I think that's the best option.

**Comment by ddanielr** [comment]
> I too am confused on what capability we lose if we implement 2. I think that's the best option.

That's my mistake, I was thinking about the optional args we allow to be passed for `accumulo-service`. While you can internally modify `accumulo-cluster` to send extra args through, you cannot pass them directly in on the command line. 

Yeah I agree that option 2 is the best approach.

**Comment by ctubbsii** [comment]
I'm including a fix for this in my updates to #5174
